JobConfig/primary/prolog.fcl

```diff
@@ -16,59 +15,75 @@ Primary: {
     g4run : @local::mu2e.physics.producers.g4run.muons
     genCounter: { module_type: GenEventCounter }
     @table::EventGenerator.producers
-    @table::CommonMC.producers
-    @table::TrackerMC.producers
-    @table::CaloDigiMC.producers
-    @table::CrvDAQPackage.producers
-# dedicated digi maker for filter
-# 2 stages now
-    makeSGS : {
-      module_type : MakeStrawGasSteps
-      KeepDeltasModule : "g4run" # don'e compress deltas on primary particles
+    @table::TrackerMC.StepProducers
+    @table::CaloMC.StepProducers
+    # the following should be reconfigured to run after step compression once that is working
+    FindMCPrimary : {
+      module_type : FindMCPrimary
+      SinglePrimary : true
+      GenParticles : "generate"
+      SimParticles : "g4run"
+      PrimaryGenIds : [] # this is intentionally empty: there should only be 1 GenParticle in primary production
     }
-    makeSD:{
-      @table::makeSD
-    }
-    compressDigiMCs : @local::DigiCompression.Primary
   }
   filters : {
 # generator truncation filter; only needed for cosmic rays in principle
     g4status: {
       module_type: FilterStatusG4
       input: "g4run"
     }
-# digi filtering for Primary selection.
-    DigiFilter : {
-      module_type : StrawDigiMCFilter
-      StrawDigiMCCollection : "makeSD"
-    }
-# shower filter for calorimeter selection
-    CaloFilter : {
-      module_type : CaloShowerSimFilter
+# step filter for Sim Primary selection
+    DetStepFilter : {
+      module_type : DetectorStepFilter
+      StrawGasSteps : [ "StrawGasStepMaker" ] 
+      CaloShowerSteps : [ "CaloShowerStepMaker" ]
+      CrvSteps : [ "g4run:CRV" ]
+      MinimumTrkStepEnergy : 1.0e-6 # 1 ev
+      MinimumCaloStepEnergy : 0.01 # need a realistic value FIXME!
+      MinimumCrvStepEnergy : 1.0e-6 # get a real value from CRV experts FIXME!
```

Comment:
Dave, imagine a track that enters through the TS opening.
Do we want to filter it out?

Reply:
If it leaves nothing that could ever be reconstructed as a signal, why not filter it out?  Note this is for primary configuration, the pileup configuration will be looser (any detectable signal).  You might also want looser selection for dedicated CRV testing, but that's not what primary is for.

Reply:
Sorry, I have not carefully looked at DetectorStepFilter, which selects on OR between tracker, calo and CRV.
In this case, I agree TS-opening cosmic events will be selected by the filter, using tracker.

Would it make sense to add a funcionaily to require AND between tracker and calorimeter?
It would be a useful feature to preselect good cosmic track cadidates.